File: 킬링캠프/2주 차 - 리스트, 큐, 스택, 그래프, DFS, BFS/practice/Keys and Rooms/dfs/Solution.java
import java.util.*;

class Solution {
    public boolean canVisitAllRooms(List<List<Integer>> rooms) {
        boolean[] visited = new boolean[rooms.size()];
        dfs(rooms, visited, 0);
        for (boolean v : visited) {
            if (!v) {
                return false;
            }
        }

        return true;   
    }

    private void dfs(List<List<Integer>> rooms, boolean[] visited, int current) {
        visited[current] = true;
        for (int next : rooms.get(current)) {
            if (visited[next]) {
                continue;
            }
            dfs(rooms, visited, next);
        }
    }
}

File: 킬링캠프/2주 차 - 리스트, 큐, 스택, 그래프, DFS, BFS/practice/Keys and Rooms/dfs/solution.py
class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        visited = [False for _ in rooms]

        visited[0] = True
        self.dfs(rooms, visited, 0)

        return all(visited)

    def dfs(self, rooms, visited, current):
        for next in rooms[current]:
            if (visited[next]):
                continue
            
            visited[next] = True
            self.dfs(rooms, visited, next)
